### PR TITLE
Move network interface speed, duplex, and operstate variables to charts

### LINKED ISSUE #10671: netdata: net: interface state: prometheus: used old metrics format
Currently we can't alert, like `interface {{ $labels.family }} on {{ $labels.instance }} is down`, because metrics use old prometheus format:

```python
netdata_net_bond0_operstate{instance="netdata"}
netdata_net_eth0_operstate{instance="netdata"}
netdata_net_vlan445_operstate{instance="netdata"}
netdata_net_tap1_operstate{instance="netdata"}
netdata_net_eth0_nic_speed_max{instance="netdata"}
```
should be:
```python
netdata_net_interface_operstate{instance="netdata", family="bond0"}
netdata_net_interface_operstate{instance="netdata", family="eth0"}
netdata_net_interface_operstate{instance="netdata", family="vlan445"}
netdata_net_interface_operstate{instance="netdata", family="tap1"}
netdata_net_interface_operstate{instance="netdata", family="eth0"}
```

### PR DESCRIPTION
**Pull request by vlvkobal**
##### Summary
To allow alerting on network interface status we are adding new charts for `speed`, `duplex`, and `operstate`. There is no need for `netdata_net_*_duplex` and `netdata_net_*_operstate` host variables so they are deleted.

Fixes #10671

##### Component Name
proc plugin

##### Test Plan
Check `http://localhost:19999/api/v1/allmetrics?format=prometheus_all_hosts&variables=yes`, you should see `netdata_net_*_nic_speed_max` variables but not `netdata_net_*_duplex` and `netdata_net_*_operstate`.

Check for new charts in the dashboard -> Network Interfaces section
```
net_speed.*
net_duplex.*
net_operstate.*
```

### CODE REVIEW

File: collectors/proc.plugin/proc_net_dev.c [hunk 1]
```diff
@@ -659,6 +693,39 @@ int do_proc_net_dev(int update_every, usec_t dt) {
             d->tcarrier    = str2kernel_uint_t(procfile_lineword(ff, l, 15));
         }
 
+        if (d->do_duplex != CONFIG_BOOLEAN_NO && d->filename_duplex) {
+            char buffer[32 + 1];
```
Comment (thiagoftsm) [inline collectors/proc.plugin/proc_net_dev.c:697]: I will give a suggestion here,  that I received in other PRs, please, instead to use `32 + 1`, add a `define` that will give a meaningful name.

File: collectors/proc.plugin/proc_net_dev.c [hunk 2]
```diff
@@ -659,6 +693,39 @@ int do_proc_net_dev(int update_every, usec_t dt) {
...
+                    d->duplex = 2;
+                else if (buffer[0] == 'h')
+                    d->duplex = 1;
+                else
+                    d->duplex = 0;
+            }
+        }
+
+        if(d->do_operstate != CONFIG_BOOLEAN_NO && d->filename_operstate) {
+            char buffer[32 + 1], *trimmed_buffer;
```
Comment (thiagoftsm) [inline collectors/proc.plugin/proc_net_dev.c:715]: If you accepts the previous suggestion, please, add it here too. Why do you have the "same variable" declared in different scopes? I think we could use only one outside these conditions. Did you receive any warning to reduce scopes?

Comment (vlvkobal) [reply collectors/proc.plugin/proc_net_dev.c]: I didn't get any warnings but I don't see the point of making the scope of the variable too wide (half of the whole code of the `proc/net` module).

**Comment by thiagoftsm** [comment]
While I was testing this PR, I observed an unexpected output, my `net_speed.wlan0` is created:

````
"net_speed.wlan0": 		{
			"id": "net_speed.wlan0",
			"name": "net_speed.wlan0",
			"type": "net_speed",
			"family": "wlan0",
			"context": "net.speed",
			"title": "Interface Speed (net_speed.wlan0)",
			"priority": 7007,
			"plugin": "proc.plugin",
			"module": "/proc/net/dev",
			"enabled": true,
			"units": "kilobits/s",
			"data_url": "/api/v1/data?chart=net_speed.wlan0",
			"chart_type": "line",
			"duration": 1,
			"first_entry": 1615382034,
			"last_entry": 1615382034,
			"update_every": 1,
			"dimensions": {
				"speed": { "name": "speed" }
			},
			"chart_variables": {
			},
			"green": null,
			"red": null,
			"alarms": {

			},
			"chart_labels": {

			}

		},
````

, but I cannot see data for it:

```
2021-03-10 13:13:54: netdata INFO  : PLUGIN[proc] : INTERNAL CHECK: host 'hades', chart 'net_speed.wlan0' is collected in the same interpolation point: short by 500000 microseconds
2021-03-10 13:19:16: netdata ERROR : WEB_SERVER[static1] : RRDR is empty for net_speed.wlan0 (RRDR has 1 dimensions, options is 0x00000223)
2021-03-10 13:19:27: netdata ERROR : WEB_SERVER[static3] : RRDR is empty for net_speed.wlan0 (RRDR has 1 dimensions, options is 0x00000223)
2021-03-10 13:19:55: netdata ERROR : WEB_SERVER[static4] : RRDR is empty for net_speed.wlan0 (RRDR has 1 dimensions, options is 0x00000223)
2021-03-10 13:19:56: netdata ERROR : WEB_SERVER[static4] : RRDR is empty for net_speed.wlan0 (RRDR has 1 dimensions, options is 0x00000223)
2021-03-10 13:19:57: netdata ERROR : WEB_SERVER[static3] : RRDR is empty for net_speed.wlan0 (RRDR has 1 dimensions, options is 0x00000223)
2021-03-10 13:21:23: netdata ERROR : WEB_SERVER[static4] : RRDR is empty for net_speed.wlan0 (RRDR has 1 dimensions, options is 0x00000223)
2021-03-10 13:21:24: netdata ERROR : WEB_SERVER[static4] : RRDR is empty for net_speed.wlan0 (RRDR has 1 dimensions, options is 0x00000223)
```

**Review by thiagoftsm — APPROVED** [review]
After the last commits, the graphs are working as expected, independent of the interface to be `wireless` or `ethernet`.

**Review by ilyam8 — APPROVED** [review]

**Pull request merged**